### PR TITLE
Tree-shaking / ES Modules support / Live-demo enhancements

### PR DESCRIPTION
**Pull request by ardatan**
- Change "target" to "es2018" in tsconfig.json
- Change "module" to "es2015" in tsconfig.json
- Change "main" to "cjs/index.js" as proxy to ES6 bundle for CJS environments like NodeJS
- Add "module": "dist/index.js" for ES6 modules
- Seperate testing module and add it to devDependencies in package.json, because Jest doesn't compile ES6 modules in node_modules. So it will just take CJS bundle of our testing scripts from `graphql-codegen-testing`

### CODE REVIEW

File: packages/graphql-codegen-cli/tsconfig.json [hunk 1]
```diff
@@ -1,8 +1,11 @@
 {
   "compilerOptions": {
+    "esModuleInterop": true,
+    "importHelpers": true,
     "experimentalDecorators": true,
-    "module": "commonjs",
-    "target": "es5",
+    "allowSyntheticDefaultImports": true,
```
[the remark this answers is not in the document]
Comment (ardatan) [reply packages/graphql-codegen-cli/tsconfig.json]: @kamilkisiela TSC was complaining on default imports, and I agree with you. I will try to figure out without adding `esModuleInterop` and `allowSyntheticDefaultImports`.

File: packages/graphql-codegen-cli/cjs/index.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+require = require('esm')(module);
```
Comment (dotansimha) [inline packages/graphql-codegen-cli/cjs/index.js]: How we can avoid replicating this file?

File: packages/old-templates/graphql-codegen-apollo-angular-template/package.json [hunk 3]
```diff
@@ -2,6 +2,8 @@
   "name": "graphql-codegen-apollo-angular-template",
```
Comment (dotansimha) [inline packages/old-templates/graphql-codegen-apollo-angular-template/package.json]: Please remove the entire directory (old-templates)

Comment (ardatan) [reply packages/old-templates/graphql-codegen-apollo-angular-template/package.json]: Done!

File: packages/graphql-codegen-cli/src/init.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
-import * as inquirer from 'inquirer';
+import inquirer from 'inquirer';
 import chalk from 'chalk';
 import { Types } from 'graphql-codegen-core';
 import { resolve, relative } from 'path';
 import { writeFileSync, readFileSync } from 'fs';
-import * as YAML from 'json-to-pretty-yaml';
-import * as detectIndent from 'detect-indent';
+import YAML from 'json-to-pretty-yaml';
```
[the remark this answers is not in the document]
Comment (ardatan) [reply packages/graphql-codegen-cli/src/init.ts]: now I use `{ stringify }`, so I think no need to worry about it

File: packages/graphql-codegen-cli/tsconfig.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ardatan) [reply packages/graphql-codegen-cli/tsconfig.json]: Removed these two options :)

**Comment by ardatan** [comment]
@dotansimha  @DAB0mB I got a weird error from `graphql-tag-pluck`. Do you have any idea?
```
lerna ERR! FAIL tests/flow.spec.ts
lerna ERR!   ● Test suite failed to run
lerna ERR! 
lerna ERR!     /home/circleci/project/node_modules/@babel/traverse/lib/scope/index.js:1
lerna ERR!     TypeError: Cannot convert undefined or null to object
lerna ERR!         at Function.keys (<anonymous>)
lerna ERR! 
lerna ERR!       at Object.<anonymous> (../../../node_modules/@babel/traverse/lib/scope/index.js:894:24)
lerna ERR!       at Object.<anonymous> (../../../node_modules/@babel/traverse/lib/path/index.js:22:37)
lerna ERR!       at Object.<anonymous> (../../../node_modules/@babel/traverse/lib/context.js:8:36)
lerna ERR!       at Object.<anonymous> (../../../node_modules/@babel/traverse/lib/index.js:27:39)
lerna ERR!       at Object.<anonymous> (../../../node_modules/graphql-tag-pluck/build/graphql-tag-pluck.js:335:18)
lerna ERR!       at __webpack_require__ (../../../node_modules/graphql-tag-pluck/build/graphql-tag-pluck.js:22:30)
lerna ERR!       at Module.<anonymous> (../../../node_modules/graphql-tag-pluck/build/graphql-tag-pluck.js:285:73)
lerna ERR!       at __webpack_require__ (../../../node_modules/graphql-tag-pluck/build/graphql-tag-pluck.js:22:30)
lerna ERR!       at Module.<anonymous> (../../../node_modules/graphql-tag-pluck/build/graphql-tag-pluck.js:105:69)
lerna ERR!       at __webpack_require__ (../../../node_modules/graphql-tag-pluck/build/graphql-tag-pluck.js:22:30)
lerna ERR!       at Object.<anonymous> (../../../node_modules/graphql-tag-pluck/build/graphql-tag-pluck.js:93:18)
```

**Pull request merged**